### PR TITLE
Conversations.replies, plus reimplement the pager of conversations.history

### PR DESCRIPTION
**Pull request by igrep**
Related: https://github.com/jpvillaisaza/slack-web/issues/82

Important commits are:

- Implement conversations.replies
    - [(WIP)(NOT WORKING) Implement conversations.replies and add a command … · jpvillaisaza/slack-web@be6e397](https://github.com/jpvillaisaza/slack-web/commit/be6e39746266bb920de7aee5f688a5a1c8697fdf)
    - [Fix: Don't add to query parameters if Nothing · jpvillaisaza/slack-web@6d1404c](https://github.com/jpvillaisaza/slack-web/commit/6d1404cc031cfc506ad0e6509cf00d125cb3f6a3)
- [Rewrite the function to get all paginated responses from conversation… · jpvillaisaza/slack-web@c153f51](https://github.com/jpvillaisaza/slack-web/commit/c153f51234e212f06481c3c18c5f11e60c7521a6)
- [Rewrite the function to get all replies in a thread · jpvillaisaza/slack-web@ac42197](https://github.com/jpvillaisaza/slack-web/commit/ac42197d90379f07e68d26ba96fdc7e604846378)
- [Test pagination of conversations.replies · jpvillaisaza/slack-web@4339ef6](https://github.com/jpvillaisaza/slack-web/commit/4339ef677ace293ada599f6afb9098ef211d7e4c)
    - [Refactor the tests · jpvillaisaza/slack-web@2482a79](https://github.com/jpvillaisaza/slack-web/commit/2482a79689f04e1d967acee81a41e7987260d354)

### CODE REVIEW

**Comment by igrep** [comment]
I've fixed the problem.

~~Note about the unverified commits: I recently updated the email address of the key `CEE2152B8AA08691`, but I shouldn't have done that. It seems that only one email address is attached to a signing key. So I must add a new subkey with~~

~~Anyway both whosekiteneverfly@gmail.com and igrep@n.email.ne.jp are my address (See https://github.com/igrep).~~

**Comment by jpvillaisaza** [comment]
Hey @igrep! Thank you for your work on the library. I invited you as collaborator of the repository so that you can merge pull requests going forward (and open them without using a fork if you prefer). I can also add you as a Hackage maintainer so you can upload new versions of the library (if this sounds good, can you give me your Hackage username?).

I'm not currently using the library, so I might be slow to review or might not keep up with all the changes. Feel free to request a review from me, but also feel free to merge your changes if I don't get back to you soon.

File: tests/Web/Slack/PagerSpec.hs
Comment (jpvillaisaza) [inline tests/Web/Slack/PagerSpec.hs:78]: Since `prepare` is called for all tests here, would it benefit from using [`before`](https://hackage.haskell.org/package/hspec-2.7.4/docs/Test-Hspec.html#v:before) or [`beforeAll`](https://hackage.haskell.org/package/hspec-2.7.4/docs/Test-Hspec.html#v:beforeAll)?

**Review by jpvillaisaza — APPROVED** [review]

**Comment by igrep** [comment]
> I can also add you as a Hackage maintainer so you can upload new versions of the library (if this sounds good, can you give me your Hackage username?).

Thank you! The user name of my Hackage account is same as GitHub: http://hackage.haskell.org/user/igrep

**Comment by jpvillaisaza** [comment]
@igrep Great! I added you as maintainer.

File: tests/Web/Slack/PagerSpec.hs
Comment (igrep) [reply tests/Web/Slack/PagerSpec.hs:78]: I personally don't like `before` etc. because functions like `prepare` here is easier to understand how it works.
Current Hspec's architecture is not suitable for RSpec's `before` etc.

Comment (jpvillaisaza) [reply tests/Web/Slack/PagerSpec.hs:78]: Sounds good.

**Review by igrep — COMMENTED** [review]
(For my TODO list)

File: src/Web/Slack.hs
Comment (igrep) [reply src/Web/Slack.hs:639]: Oops, this is actually used by the existing functions. I think we should deprecate the old paginators because both `mpimHistory` and `channelsHistory` will be unavailable due to the coming updates of Slack API.
But I'll do it later anyway.

File: src/Web/Slack.hs
Comment (igrep) [reply src/Web/Slack.hs:597]: Same as https://github.com/jpvillaisaza/slack-web/pull/84/files/edbc1c7d88615f5b5eaf255a5915218c9e7d74ed#r528668589

**Comment by igrep** [comment]
After merging this pull request, I'll use the development version for https://github.com/haskell-jp/slack-log to find some points to improve. Then, I want to deprecate (or delete) the old APIs which is going to be shut down by https://api.slack.com/changelog/2020-01-deprecating-antecedents-to-the-conversations-api, before releasing the new version.

**Pull request merged**
